src/ScriptEditor/ui/ScriptEditorRoot.tsx

```diff
@@ -187,19 +255,40 @@ function Root(props: IProps): React.ReactElement {
     debouncedCodeParsing(newCode);
   };
 
+  let changeModelCallback: monaco.IDisposable | null = null;
+
   // When the editor is mounted
   function onMount(editor: IStandaloneCodeEditor): void {
     // Required when switching between site navigation (e.g. from Script Editor -> Terminal and back)
     // the `useEffect()` for vim mode is called before editor is mounted.
     editorRef.current = editor;
+    changeModelCallback = editorRef.current.onDidChangeModel((e) => {
+      if (e.newModelUrl === null) {
+        return;
+      }
+
+      const resolvedPath = resolveScriptFilePath(e.newModelUrl.path);
+      if (resolvedPath === null) {
+        return false;
+      }
+      const [hostname, path] = resolvedPath.split("/", 2);
+
+      const script = openScripts.find((s) => s.hostname === hostname && s.path === path);
+      if (script) {
+        parseCode(script.code);
+      }
+
+      rerender();
+      removeOutlineOfEditor();
+    });
+    vim = props.vim;
```

Comment:
Why is this changing?

Reply:
When using code navigation, monaco might open a file that has not been loaded yet. Originally, `props.vim` is set by the parent ReactNode when calling OnMount; unfortunately, the CodeEditorOpener doesn't know the props and hence I save the most recent value so that new files are opened with the latest `props.vim` setting instead of falling back to a fixed `true` or `false`.